src/debug.h

```diff
@@ -30,18 +30,9 @@ void *alloca(size_t);
 
 void Debug(const char *str, ...);
 
-#ifdef HAVE_ALLOCA
-
-#   define AllocateStack( x ) alloca( x )
-#   define ReleaseStack( x ) ((void)0)
-
-#else
-
```

Comment:
What's the problem with `alloca`? This forces all allocations to be heap allocations.  I'd be interested to see what the performance difference is on a slow system.

Reply:
alloca may be dangerous.

https://man7.org/linux/man-pages/man3/alloca.3.html
https://stackoverflow.com/questions/1018853/why-is-the-use-of-alloca-not-considered-good-practice